BSP/Inc/BSP_Lights.h

```diff
@@ -0,0 +1,33 @@
+#ifndef __BSP_Lights_H
+#define __BSP_Lights_H
+
+#include <bsp.h>
+#include "common.h"
+#include "config.h"
+
+
+typedef enum {M_CNCTR=0,RSVD_LED,CTRL_FAULT,RIGHT_BLINK,HeadlightPWR,A_CNCTR,BPS_FAULT,LEFT_BLINK,BPS_PWR,BrakeLight} LIGHT_t;
+
+/**
+* @brief   Initialize Lights Module
+* @return  void
+*/ 
+void BSP_Lights_Init();
+
+/**
+* @brief   Read the state of the lights
+* @param   LightChannel Which Light to read
+* @return  returns state enum which indicates ON/OFF
+*/ 
+state_t BSP_Lights_Read(LIGHT_t LightChannel);
+
+/**
+* @brief   Switch light between ON/OFF states
+* @param   LightChannel Which light to switch
+* @return  void
+*/ 
+void BSP_Lights_Switch(LIGHT_t LightChannel);
```

Comment:
We don't always want to toggle the light, so I would make this function called `BSP_Lights_Set()` and also take in a `state_t` parameter. That way we don't need to know whether or not the lights are on to turn them on or off to turn them off. Mainly it'll just relieve us of one step in the application layer.